### PR TITLE
Settings: fix Tracks event when enabling/disabling advanced features settings.

### LINKED ISSUE #32294: Analytics Setting: The reload-page hook should not be added in the result of the toggle setting response
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

While working on [updating the analytics settings](<https://github.com/woocommerce/woocommerce-ios/pull/6457>) from the Woo Mobile app, I found that the request to update the analytics setting always fails when I toggle the value from `false` to `true` with an error.

This issue doesn't happen when calling the API directly, but only when calling it through Jetpack tunnel, so I asked the Jetpack Crew team and it seems that the reload-page hook is the reason:

> When the setting is toggled, there’s a [hook to reload the page ](<https://github.com/woocommerce/woocommerce/blob/c4628b966be16218a99d093cf3672ad79a4f235e/plugins/woocommerce/src/Internal/Admin/Analytics.php#L89>)which is causing additional requests to be made before returning a REST response.<br>Those end up being GET requests with a non-empty body, as suggested above, causing the aforementioned error.<br>Removing the hook solves the issue.

I wonder if there's a different way to reload the page on the frontend and remove the hook on the backend side. That will avoid causing the false failure in the Jetpack tunneled requests to the API.

Internal reference: p1647845508355109-slack-CBG1CP4EN

### Expected behavior

The Jetpack tunneled request to update the analytics setting does not return the error.

### Actual behavior

Jetpack tunneled requests to update the `woocommerce_analytics_enabled` setting fail ONLY when switching from “yes” to “no” OR “no” to “yes”, with the error:  `"This request method does not support body parameters."`

### Steps to reproduce

Call the API through Jetpack tunneled to toggle the value of the analytics setting. Sample CURL:

```
curl --location --request POST 'https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/<blog-id>/rest-api' \
--header 'Authorization: <Token>' \
--header 'Accept: application/json' \
--form 'path="wc/v3/settings/advanced/woocommerce_analytics_enabled"' \
--form 'body="{ \"value\": \"yes\" }"' \
--form '_method="put"' \
--form 'json="true"'
```

Notice that if the value is previously "no", toggling the value to "true" with the above CURL will result in a 400 error:

```
{
    "error": "rest_invalid_request",
    "message": "This request method does not support body parameters."
}
```

### WordPress Environment

Click to expand!

WordPress address (URL): [https://huongdotests1.blog](<https://huongdotests1.blog>)<br>Site address (URL): [https://huongdotests1.blog](<https://huongdotests1.blog>)<br>WC Version: 6.3.1<br>REST API Version: ✔ 6.3.1<br>WC Blocks Version: ✔ 6.9.0<br>Action Scheduler Version: ✔ 3.4.0<br>WC Admin Version: ✔ 3.2.1<br>Log Directory Writable: ✔<br>WP Version: 5.9.2<br>WP Multisite: –<br>WP Memory Limit: 512 MB<br>WP Debug Mode: ✔<br>WP Cron: ✔<br>Language: en_US<br>External object cache: ✔

### Server Environment

Server Info: nginx<br>PHP Version: 7.4.28<br>PHP Post Max Size: 2 GB<br>PHP Time Limit: 1200<br>PHP Max Input Vars: 6144<br>cURL Version: 7.82.0<br>OpenSSL/1.1.1d

SUHOSIN Installed: –<br>MySQL Version: 5.5.5-10.4.24-MariaDB-log<br>Max Upload Size: 2 GB<br>Default Timezone is UTC: ✔<br>fsockopen/cURL: ✔<br>SoapClient: ✔<br>DOMDocument: ✔<br>GZip: ✔<br>Multibyte String: ✔<br>Remote Post: ✔<br>Remote Get: ✔

### Database

WC Database Version: 6.3.1<br>WC Database Prefix: wp\_<br>Total Database Size: 12.88MB<br>Database Data Size: 9.19MB<br>Database Index Size: 3.69MB<br>wp_woocommerce_sessions: Data: 0.08MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_woocommerce_order_items: Data: 0.05MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_order_itemmeta: Data: 0.16MB + Index: 0.16MB + Engine InnoDB<br>wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_actionscheduler_actions: Data: 0.16MB + Index: 0.16MB + Engine InnoDB<br>wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_actionscheduler_logs: Data: 0.09MB + Index: 0.09MB + Engine InnoDB<br>wp_ce4wp_abandoned_checkout: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_comments: Data: 0.13MB + Index: 0.09MB + Engine InnoDB<br>wp_gla_budget_recommendations: Data: 0.22MB + Index: 0.14MB + Engine InnoDB<br>wp_gla_merchant_issues: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_gla_shipping_rates: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_mailpoet_mapping_to_external_entities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB<br>wp_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB<br>wp_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_subscribers: Data: 0.02MB + Index: 0.13MB + Engine InnoDB<br>wp_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_ms_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_options: Data: 1.09MB + Index: 0.06MB + Engine InnoDB<br>wp_postmeta: Data: 1.48MB + Index: 0.38MB + Engine InnoDB<br>wp_posts: Data: 1.30MB + Index: 0.13MB + Engine InnoDB<br>wp_snippets: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB<br>wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB<br>wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB<br>wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB<br>wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_wdp_orders: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_wdp_order_items: Data: 0.02MB + Index: 0.06MB + Engine InnoDB<br>wp_wdp_rules: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_yith_wcwl: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_yith_wcwl_lists: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_yoast_indexable: Data: 0.20MB + Index: 0.09MB + Engine InnoDB<br>wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB<br>wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB<br>wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB<br>wp_yoast_seo_links: Data: 0.02MB + Index: 0.03MB + Engine InnoDB

### Post Type Counts

amp_validated_url: 2<br>attachment: 149<br>customize_changeset: 5<br>feedback: 17<br>jetpack_migration: 2<br>jp_img_sitemap: 6<br>jp_sitemap: 6<br>jp_sitemap_master: 6<br>mailpoet_page: 1<br>page: 9<br>post: 2<br>product: 19<br>product_variation: 5<br>revision: 23<br>shop_coupon: 31<br>shop_order: 208<br>shop_order_refund: 4<br>wp_global_styles: 1<br>wp_template_part: 4

### Security

Secure connection (HTTPS): ✔<br>Hide errors from visitors: ✔

### Active Plugins (18)

WCS staging: by mb – 1.0<br>Akismet Anti-Spam: by Automattic – 4.2.2<br>AMP: by AMP Project Contributors – 2.2.1<br>CoBlocks: by GoDaddy – 2.19.1<br>Crowdsignal Forms: by Automattic – 1.6.0<br>[WordPress.com](<http://WordPress.com>) Editing Toolkit: by Automattic – 3.29347<br>Google Listings and Ads: by WooCommerce – 1.11.1<br>Gutenberg: by Gutenberg Team – 12.8.1<br>Jetpack: by Automattic – 10.8-a.7<br>Layout Grid: by Automattic – 1.8.1<br>Page Optimize: by Automattic – 0.5.2<br>WooCommerce Beta Tester: by WooCommerce – 2.0.5<br>WooCommerce Stripe Gateway: by WooCommerce – 6.3.0<br>WooCommerce Payments Dev Tools: by Automattic –<br>WooCommerce Payments: by Automattic – 3.8.2<br>WooCommerce Shipping & Tax: by WooCommerce – 1.25.24<br>WooCommerce: by Automattic – 6.3.1<br>Yoast SEO: by Team Yoast – 18.3

### Inactive Plugins (7)

Checkout Fields Manager for WooCommerce: by QuadLayers – 5.5.5<br>Classic Editor: by WordPress Contributors – 1.6.2<br>Code Snippets: by Code Snippets Pro – 2.14.3<br>Creative Mail by Constant Contact: by Constant Contact – 1.5.0<br>Mailchimp for WooCommerce: by Mailchimp – 2.6.1<br>MailPoet 3 (New): by MailPoet – 3.84.1<br>WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.1.3

### Dropin Plugins (2)

advanced-cache.php: advanced-cache.php<br>object-cache.php: Memcached

### Must Use Plugins (1)

[WP.com](<http://WP.com>) Site Helper: by  –

### Settings

API Enabled: –<br>Force SSL: –<br>Currency: USD ($)<br>Currency Position: left<br>Thousand Separator: ,<br>Decimal Separator: .<br>Number of Decimals: 2<br>Taxonomies: Product Types: external (external)<br>grouped (grouped)<br>simple (simple)<br>subscription (subscription)<br>variable (variable)<br>variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)<br>exclude-from-search (exclude-from-search)<br>featured (featured)<br>outofstock (outofstock)<br>rated-1 (rated-1)<br>rated-2 (rated-2)<br>rated-3 (rated-3)<br>rated-4 (rated-4)<br>rated-5 (rated-5)

Connected to [WooCommerce.com](<http://WooCommerce.com>): –

### WC Pages

Shop base: #16 - /shop/<br>Cart: #17 - /cart/<br>Checkout: #18 - /checkout/<br>My account: #19 - /my-account/<br>Terms and conditions: ❌ Page not set

### Theme

Name: Storefront<br>Version: 4.0.0<br>Author URL: [https://woocommerce.com/](<https://woocommerce.com/>)<br>Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme<br>WooCommerce Support: ✔

### Templates

Overrides: –

### Subscriptions

WCS_DEBUG: ✔ No<br>Subscriptions Mode: ✔ Live<br>Subscriptions Live URL: [https://huongdotests1.blog](<https://huongdotests1.blog>)<br>Subscription Statuses: –<br>WooCommerce Account Connected: ❌ No

### Store Setup

Country / State: United States (US) — California

### Payment Gateway Support

WooCommerce Payments: products<br>refunds<br>multiple_subscriptions<br>subscription_cancellation<br>subscription_payment_method_change_admin<br>subscription_payment_method_change_customer<br>subscription_payment_method_change<br>subscription_reactivation<br>subscription_suspension<br>subscriptions<br>gateway_scheduled_payments<br>tokenization<br>add_payment_method

Cash on delivery: products

### WooCommerce Payments

Version: 3.8.2<br>Connected to WPCOM: Yes<br>Blog ID: 194373765<br>Account ID: acct_1JUqHZ2EcaTRX0Wr

### Action Scheduler

Canceled: 3<br>Oldest: 2022-02-23 17:11:20 +0700<br>Newest: 2022-03-03 10:45:02 +0700

Complete: 403<br>Oldest: 2022-02-18 18:31:29 +0700<br>Newest: 2022-03-21 14:17:25 +0700

Failed: 1<br>Oldest: 2021-08-05 11:22:12 +0700<br>Newest: 2021-08-05 11:22:12 +0700

Pending: 3<br>Oldest: 2022-03-21 19:10:56 +0700<br>Newest: 2022-04-05 11:06:09 +0700

### Status report information

Generated at: 2022-03-21 18:03:02 +07:00

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](<https://woocommerce.com/storefront/>).
- [X] I can reproduce this bug consistently using the steps above.

### LINKED ISSUE #33128: The `wcadmin_settings_change` Tracks event isn't being logged correctly for `settings: woocommerce_navigation_enabled`
### Describe the bug

When the Navigation feature is toggled in Settings, the `wcadmin_settings_change` Tracks event is supposed to be logged with `settings: woocommerce-navigation_enabled`.

Screenshot%20on%202022-05-19%20at%2012%3A26%3A14
 
However, looking at our analytics, this does not appear to be working correctly.

The Tracks event introduced when the user enables/disables the navigation (
WooCommerce - Settings - Advanced - Features) is broken. This event should be fixed in order to understand how many users are enabling or disabling the navigation. And the same should be applied to the analytics feature.

### Expected behavior

The `wcadmin_settings_change` Tracks event is supposed to be logged with `settings: woocommerce-navigation_enabled`.

### Actual behavior

The `wcadmin_settings_change` Tracks event is not being logged with `settings: woocommerce-navigation_enabled`

### Acceptance criteria

- [ ] Fire `wcadmin_settings_change` (settings: woocommerce_navigation_enabled) when the user enables the navigation
- [ ] Fire `wcadmin_settings_change` (settings: woocommerce_navigation_disabled) when the user disables the navigation
- [ ] Fire `wcadmin_settings_change` (settings: woocommerce_analytics_enabled) when the user enables analytics
- [ ] Fire `wcadmin_settings_change` (settings: woocommerce_analytics_disabled) when the user disables analytics

### References

- Slack thread: p1652965397899679-slack-C01CMSA1VHS

### PR DESCRIPTION
**Pull request by AnnaMag**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Track events of user actions enabling/disabling the navigation and analytics settings (WooCommerce - Settings - Advanced - Features) is not recording any data atm, while we need it for usage statistics.

Here, we fix the event `wcadmin_settings_change` with **property** `settings` of ` woocommerce_navigation_enabled`/` woocommerce_analytics_enabled` to fire on checkbox settings changes.
The corresponding **options** are `woocommerce_navigation_enabled` and `woocommerce_navigation_disabled` with yes/no values.

In a follow up we will expand the event schema to distinguish enable/disable actions.
Closes #33128, closes #32294, closes WOOPLUG-642

### How to test the changes in this Pull Request:

1. go to `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. enable/disable de settings
3. verify in Tracks that the appropriate event was fired
Screenshot 2022-06-03 at 11 28 59
Note: for testing in a set-up where `userid = null`, the events will show as rejected in Tracks live

4. verify that the relevant option value is set correctly

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33305/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Remove feature specific code in tracks class` 7ef31da36a386ca481b0e9b0eccdc6c39015cce1
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php [hunk 1]
```diff
@@ -78,7 +78,24 @@ public function track_setting_change( $option_name, $old_value, $new_value ) {
 			return;
 		}
 
+		// This is not used for site option storing/updates,
+		// which are yes/no *_enabled*.
+		// It is merely setting up $this->updated_options[] array
+		// with property names used in tracking events.
+		if ( 'woocommerce_analytics_enabled' === $option_name && 'no' === $new_value ) {
```
Comment (joshuatf) [inline plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php]: Could these be added dynamically inside the `Features` class in the `enable|disable` methods?

https://github.com/woocommerce/woocommerce/blob/21f7cea2b6ab1a268c137dcf761c4df721f1ec0b/plugins/woocommerce/src/Admin/Features/Features.php#L198-L230

**Review by joshuatf — COMMENTED** [review]
Nice sleuthing, @AnnaMag!  Just left a comment about possibly moving this into the features namespace.

**Comment by joshuatf** [comment]
The root cause of this was early exiting on the `update_option_` hook.

https://github.com/woocommerce/woocommerce/blob/3fd9ee99aa72b9dc1524ab7b2a8be06add53161f/plugins/woocommerce/src/Internal/Admin/Analytics.php#L83-L98

This never allowed the wc settings tracks changes to be run since they were on a later hook.

https://github.com/woocommerce/woocommerce/blob/3fd9ee99aa72b9dc1524ab7b2a8be06add53161f/plugins/woocommerce/includes/admin/class-wc-admin-settings.php#L78-L80

**Comment by joshuatf** [comment]
The root cause of this was early exiting on the `update_option_` hook.

https://github.com/woocommerce/woocommerce/blob/3fd9ee99aa72b9dc1524ab7b2a8be06add53161f/plugins/woocommerce/src/Internal/Admin/Analytics.php#L83-L98

This never allowed the wc settings tracks changes to be run since they were on a later hook.

https://github.com/woocommerce/woocommerce/blob/3fd9ee99aa72b9dc1524ab7b2a8be06add53161f/plugins/woocommerce/includes/admin/class-wc-admin-settings.php#L78-L80

**Review by joshuatf — APPROVED** [review]
This fixes the underlying issue with tracks.  Thanks for all the discussion on this, Anna!

We'll follow-up with a future PR to address adding in the values for toggled checkboxes/features.

**Pull request merged**